benchmarking/prototype_parameterized_template.py

```
@@ -0,0 +1,613 @@
+        ]:
+            raise RuntimeError("Constant type signatures do not align")
+
+    varying = []
+    for index in range(len(reference["constants"])):
+        values = [np.asarray(item["constants"][index].data) for item in prepared]
+        first = values[0]
+        if not all(np.array_equal(first, value, equal_nan=True) for value in values[1:]):
+            if not np.issubdtype(first.dtype, np.floating):
+                raise RuntimeError(
+                    f"Varying non-floating constant at index {index}: {first.dtype}"
+                )
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)